os/overriding-systemd-units.md

```diff
@@ -0,0 +1,105 @@
+# Overriding systemd units
+
+There are two methods of overriding default CoreOS settings in unit files: copying the unit file from /usr/lib64/systemd/system to /etc/systemd/system and modifying the chosen settings. Alternatively, one can create a directory named unit.d/ within /etc/systemd/system and place a drop-in file name.conf there that only changes the specific settings one is interested in. Note that multiple such drop-in files are read if present.
+
+The advantage of the first method is that one easily overrides the complete unit, the default CoreOS unit is not parsed at all anymore. It has the disadvantage that improvements to the unit file supplied by CoreOS are not automatically incorporated on updates.
+
+The advantage of the second method is that one only overrides the settings one specifically wants, where updates to the original CoreOS unit automatically apply. This has the disadvantage that some future CoreOS updates might be incompatible with the local changes.
```

Comment:
Maybe add something about cloudinit:

"CoreOS configuration parameters included in cloud-config, such as `etcd2` settings are generated into drop-in units automatically. To view the drop-ins that are generated, use `systemctl`:

```
$ systemctl cat etcd2
# /usr/lib64/systemd/system/etcd2.service
[Unit]
Description=etcd2
Conflicts=etcd.service

[Service]
User=etcd
Environment=ETCD_DATA_DIR=/var/lib/etcd2
Environment=ETCD_NAME=%m
ExecStart=/usr/bin/etcd2
Restart=always
RestartSec=10s
LimitNOFILE=40000

# /run/systemd/system/etcd2.service.d/10-oem.conf
[Service]
Environment=ETCD_ELECTION_TIMEOUT=1200
# /run/systemd/system/etcd2.service.d/20-cloudinit.conf
[Service]
Environment="ETCD_ADVERTISE_CLIENT_URLS=http://10.230.19.235:2379,http://10.230.19.235:4001"
Environment="ETCD_DISCOVERY=https://discovery.etcd.io/c6a991b4-8d96-46cd-b36f-5a9fdaa880a4"
Environment="ETCD_INITIAL_ADVERTISE_PEER_URLS=http://10.230.19.235:2380"
Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379,http://0.0.0.0:4001"
Environment="ETCD_LISTEN_PEER_URLS=http://10.230.19.235:2380"
```

There are two drop-in units being applied. The first, `10-oem.conf` comes from the OEM, which are a few specific tweaks to CoreOS images to integrate with certain platforms like EC2. In this case, the election timeout has been raised since this image runs in the cloud with a more variable network. The second drop-in, `20-cloudinit.conf`, has been generated based on settings that we pass in via our cloud-config file. The number at the beginning of the file specifies the order in which the drop-ins are applied.